h/api/models.py

```diff
@@ -166,6 +175,113 @@ def title(self):
             return ""
 
     @property
+    def filename(self):
+        if is_file(self.uri):
+            return self.uri.split("/")[-1] or ""
+        else:
+            return ""
+
+    @property
+    def document_link(self):
```

Comment:
This is starting to look increasingly like presenter code, and it makes me a bit uneasy to have it in the model. I think for now we merge this as it is, but in future I think we should perhaps consider having an `AnnotationPresenter` class which takes an `Annotation` in its constructor and is responsible for all the mapping between the data model and the presentation model. This would presumably also make the logic in these things a bunch easier to test as it wouldn't require setting up the database connection, etc.